src/runtime/server/api/indexed-search.ts

```diff
@@ -0,0 +1,32 @@
+import { defineEventHandler } from 'h3'
+import MiniSearch from 'minisearch'
+import { splitPageIntoSections } from '../search'
+import { useRuntimeConfig } from '#imports'
+import { serverQueryContent } from '#content/server'
+
+export default defineEventHandler(async (event) => {
+  const runtimeConfig = useRuntimeConfig()
+  const { ignoredTags, ignoreDrafts, ignorePartials, ignoreEmpty, options } = runtimeConfig.public.content.search
+
+  const files = await serverQueryContent(event).find()
+
+  // Only works for MD
+  const sections = (await Promise.all(
+    files
+      .filter(file => file._extension === 'md' &&
+      (!ignoreDrafts ? true : !file?._draft) &&
+      (!ignoreEmpty ? true : !file?._empty) &&
+      (!ignorePartials ? true : !file?._partial)
```

Comment:
Could we also support some kind of `ignoreNavigationFalse`?

When a page has:
```md
---
navigation: false
---
```

Reply:
Additionally, What do you think about allowing user to define custom query instead of multiple boolean options?

As an example, if we want to search inside a specific directory (let's say `/content/docs`) we can define a custom query in module options and filter contents in side the directory.

Reply:
I like this idea yes

Reply:
me too! way better than 100 boolean